src/components/data/toolbar/DataDotMenu.js

```diff
@@ -0,0 +1,138 @@
+                            </ListItemIcon>
+                            <ListItemText primary={getMessage("details")} />
+                        </MenuItem>
+                    ),
+                    <Divider
+                        key={build(baseId, ids.DETAILS_MENU_ITEM_DIVIDER)}
+                    />,
+                    isWritable(permission) && (
+                        <MenuItem
+                            key={build(baseId, ids.CREATE_FOLDER_MI)}
+                            id={build(baseId, ids.CREATE_FOLDER_MI)}
+                            onClick={() => {
+                                onClose();
+                                onCreateFolderClicked();
+                            }}
+                        >
+                            <ListItemIcon>
+                                <CreateNewFolder fontSize="small" />
+                            </ListItemIcon>
+                            <ListItemText primary={getMessage("folder")} />
+                        </MenuItem>
+                    ),
+                    deleteMiEnabled && (
+                        <MenuItem
+                            key={build(baseId, ids.DELETE_MENU_ITEM)}
+                            id={build(baseId, ids.DELETE_MENU_ITEM)}
+                            onClick={() => {
+                                onClose();
+                                onDeleteSelected();
+                            }}
+                        >
+                            <ListItemIcon>
+                                <DeleteIcon fontSize="small" />
+                            </ListItemIcon>
+                            <ListItemText primary={getMessage("delete")} />
+                        </MenuItem>
+                    ),
+                    <Divider
+                        key={build(baseId, ids.UPLOAD_MENU_ITEM_DIVIDER)}
+                    />,
+                    <UploadMenuItems
+                        key={build(baseId, ids.UPLOAD_MENU_ITEM)}
+                        localUploadId={localUploadId}
+                        uploadMenuId={uploadMenuId}
+                        onBrowseLocal={onClose}
+                        onImportFromURL={() => {
+                            onClose();
+                            console.log("Import from URL");
+                        }}
+                        onUploadQueue={() => {
+                            onClose();
+                            setUploadDialogOpen(true);
+                        }}
+                    />,
+                ]}
+            />
+            <CreateFolderDialog
+                path={path}
+                open={createFolderDlgOpen}
+                onClose={onCreateFolderDlgClose}
+                onFolderCreated={() => {
+                    onCreateFolderDlgClose();
+                    refreshListing();
+                }}
+            />
+        </>
+    );
+}
+
+export default withI18N(DataDotMenu, messages);
```

Comment:
I have removed menuItems that have not been implemented yet to help testing / clean up.